src/EnhancedEntityRevisionTrait.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\src\EnhancedEntityRevisionTrait.
+ */
+
+namespace Drupal\src;
+
+/**
+ * Provides a trait implementing \Drupal\entity\EnhanceredEntityRevisionInterface.
+ */
+trait EnhancedEntityRevisionTrait {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRevisionCreationTime() {
+    return $this->revision_create->value;
```

Comment:
The trait should use hasField() to avoid errors if no field with that name exists in all these methods. Maybe set should throw an exception but then we need a way to check if an entity has a revision creation time and so on?

How do we make node implement this interface? Switch out the entity class?